disnake/interactions/application_command.py

```diff
@@ -33,15 +33,18 @@
 MISSING = utils.MISSING
 
 if TYPE_CHECKING:
+    from ..client import Client
     from ..ext.commands import InvokableApplicationCommand
     from ..state import ConnectionState
     from ..types.interactions import (
         ApplicationCommandInteraction as ApplicationCommandInteractionPayload,
         ApplicationCommandInteractionData as ApplicationCommandInteractionDataPayload,
     )
 
+BotT = TypeVar("BotT", bound="Client")
```

Comment:
random comment: the naming might be a bit weird here considering the bound is `Client`, not `Bot`, but for consistency's sake and since it's certainly the much more common usecase, `BotT` is the better choice here